### PR TITLE
feat: create product endpoint

### PR DESCRIPTION
**Pull request by Juninhotech**
# Description
Created endpoint to add product

### Link to issue ticket
#84 

## What were you told to do?
To create and endpoint for adding product

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
image

### CODE REVIEW

**Comment by chimaihueze** [comment]
You have conflicting files. Please resolve.

File: src/Hng.Application/Test/CreateProductTest.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+using Xunit;
+using Hng.Application.Features.Products.Handlers;
+using Hng.Application.Features.Products.Commands;
+using Hng.Domain.Entities;
+using Hng.Application.Features.Products.Dtos;
+using Hng.Infrastructure.Repository.Interface;
+
+namespace Hng.Application.UnitTests.Features.Products.Handlers
+{
+    public class CreateProductHandlerTests
```
Comment (NdubuisiJr) [inline src/Hng.Application/Test/CreateProductTest.cs:14]: Why is your test here?

File: src/Hng.Web/Controllers/ProductController.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using Hng.Application.Features.Products.Commands;
+using Hng.Application.Features.Products.Dtos;
+using Hng.Application.Features.UserManagement.Dtos;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [ApiController]
```
Comment (NdubuisiJr) [inline src/Hng.Web/Controllers/ProductController.cs:9]: Protect this route

File: src/Hng.Web/appsettings.json [hunk 3]
```diff
@@ -6,11 +6,11 @@
     }
   },
   "ConnectionStrings": {
-    "DefaultConnectionString": ""
+    "DefaultConnectionString": "Host=localhost;Database=HNGTASK;Username=postgres;Password=password1;"
```
Comment (NdubuisiJr) [inline src/Hng.Web/appsettings.json:9]: Remove your local db connection

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Fix comments and merge conflicts

**Pull request closed without merging**